crt_portal/cts_forms/views.py

```diff
@@ -182,6 +183,38 @@ def _related_reports_count(report):
         return 1
 
 
+def _get_email(report):
+    return report.contact_email
+
+
+def _set_constant_writer(contact_templates, report):
+    for contact_template in contact_templates:
+        if contact_template[1].upper() == report.contact_email.upper():
+            report.constant_writer = True
+
+
+def _was_sent_constant_writer_email(reports):
+    report_emails = tuple(map(_get_email, reports))
```

Comment:
```suggestion
    report_emails = tuple(_get_email(r) for r in reports))
```

style nit: using the more "pythonic" comprehensions (list, set, and dict) is generally preferred over map, filter, etc

though I really appreciate the functional approach!

https://stackoverflow.com/a/1247490/13231999